src/PhonePush/IOS.hs

```
@@ -80,6 +82,15 @@ checkFailLive = checkFail feedbackLive
 checkFailTest :: FilePath -> FilePath -> IO [B.ByteString]
 checkFailTest = checkFail feedbackTest
 
+withSocketSafe :: ProtocolNumber -> (Socket -> IO a) -> IO a
+withSocketSafe proto =
+  bracket (socket AF_INET Stream proto) $ \sock ->
+    catch (close sock) $ \(e :: SomeException) ->
+      putStrLn . unwords $ [ "Caught exception trying to close"
+                           , "Apple push notifications socket:"
+                           , show e
+                           ]
```

Let's not catch all exceptions unconditionally here.  Is it actually necessary to catch exceptions coming from `close sock`?